Lib/test/test_pty.py

```diff
@@ -43,6 +54,12 @@ def normalize_output(data):
 
     return data
 
+def _readline(fd):
```

Comment:
I'm not sure you still need a helper function. Just use the `FileIO` object directly from the test method.

Also, if you remove `closefd=False`, you don't need the `os.close(master_fd)` at the end.

Reply:
The idea of keeping the helper function was to have it close to the comment I added above. And the comment is both about `_readline` and `normalize_output`.

We have `os.close(slave_fd)`. I think it would be very asymmetric if we don't `close(master_fd)` as well in the code explicitly?